code/controllers/subsystems/ghostroles.dm

```diff
@@ -177,3 +185,25 @@
 			for(var/i in S.spawnpoints)
 				update_spawnpoint_status_by_identifier(i)
 	return
+
+/datum/controller/subsystem/ghostroles/proc/add_spawn_atom(var/ghost_role_name, var/atom/spawn_atom)
+	if(ghost_role_name && spawn_atom)
+		var/datum/ghostspawner/G = spawners[ghost_role_name]
+		if(G)
+			G.spawn_atoms += spawn_atom
+			if(length(G.spawn_atoms) == 1)
+				G.enable()
+
+/datum/controller/subsystem/ghostroles/proc/remove_spawn_atom(var/ghost_role_name, var/atom/spawn_atom)
+	if(ghost_role_name && spawn_atom)
+		var/datum/ghostspawner/G = spawners[ghost_role_name]
+		if(G)
+			G.spawn_atoms -= spawn_atom
+			if(!length(G.spawn_atoms))
+				G.disable()
+
+/datum/controller/subsystem/ghostroles/proc/get_spawn_atoms(var/ghost_role_name)
+	var/datum/ghostspawner/G = spawners[ghost_role_name]
+	if(G)
+		return G.spawn_atoms
+	return list()
```

Comment:
this is going to lead to a lot of awkwardness if you return an empty list, because an empty list technically exists
```suggestion
```

Reply:
Eh, consistent return types are better, leads to less null errors. A list is typically used in a loop or otherwise, which behave as expected in the case of empty lists.